kafka/manager.go

```diff
@@ -0,0 +1,204 @@
+		errs = append(errs, err)
+	}
+	if cfg.PartitionCount == 0 {
+		errs = append(errs, errors.New("kafka: partition count must be non-zero"))
+	}
+	return errors.Join(errs...)
+}
+
+// Manager manages Kafka topics.
+type Manager struct {
+	cfg          ManagerConfig
+	topicConfigs map[string]*string
+	client       *kadm.Client
+	tracer       trace.Tracer
+}
+
+// NewManager returns a new Manager with the given config.
+func NewManager(cfg ManagerConfig) (*Manager, error) {
+	if err := cfg.Validate(); err != nil {
+		return nil, fmt.Errorf("kafka: invalid manager config: %w", err)
+	}
+	var topicConfigs map[string]*string
+	if len(cfg.TopicConfigs) != 0 {
+		topicConfigs = make(map[string]*string, len(cfg.TopicConfigs))
+		for k, v := range cfg.TopicConfigs {
+			topicConfigs[k] = kadm.StringPtr(v)
+		}
+	}
+	client, err := cfg.newClient()
+	if err != nil {
+		return nil, fmt.Errorf("kafka: failed creating kafka client: %w", err)
+	}
+	return &Manager{
+		cfg:          cfg,
+		topicConfigs: topicConfigs,
+		client:       kadm.NewClient(client),
+		tracer:       cfg.tracerProvider().Tracer("kafka"),
+	}, nil
+}
+
+// Close closes the manager's resources, including its connections to the
+// Kafka brokers and any associated goroutines.
+func (m *Manager) Close() error {
+	m.client.Close()
+	return nil
+}
+
+// CreateTopics creates one or more topics.
+//
+// Topics that already exist will be left unmodified.
+func (m *Manager) CreateTopics(ctx context.Context, topics ...apmqueue.Topic) error {
+	// TODO(axw) how should we record topics?
+	ctx, span := m.tracer.Start(ctx, "CreateTopics", trace.WithAttributes(
+		semconv.MessagingSystemKey.String("kafka"),
+	))
+	defer span.End()
+
+	topicNames := make([]string, len(topics))
+	for i, topic := range topics {
+		topicNames[i] = string(topic)
+	}
+	responses, err := m.client.CreateTopics(
+		ctx,
+		int32(m.cfg.PartitionCount),
+		-1, // default.replication.factor
+		m.topicConfigs,
+		topicNames...,
+	)
+	if err != nil {
+		return fmt.Errorf("failed to create kafka topics: %w", err)
+	}
+	createTopicParamsFields := []zap.Field{
+		zap.Int("partition_count", m.cfg.PartitionCount),
+	}
+	if m.cfg.TopicConfigs != nil {
+		createTopicParamsFields = append(createTopicParamsFields,
+			zap.Reflect("topic_configs", m.cfg.TopicConfigs),
+		)
+	}
+	var createErrors []error
+	for _, response := range responses.Sorted() {
+		logger := m.cfg.Logger.With(zap.String("topic", response.Topic))
+		if err := response.Err; err != nil {
+			if errors.Is(err, kerr.TopicAlreadyExists) {
+				logger.Debug("kafka topic already exists")
+			} else {
+				span.RecordError(err)
+				createErrors = append(createErrors,
+					fmt.Errorf(
+						"failed to create topic %q: %w",
+						response.Topic, err,
+					),
+				)
+			}
+			continue
+		}
+		logger.Info("created kafka topic", createTopicParamsFields...)
+	}
+	if err := errors.Join(createErrors...); err != nil {
+		return err
+	}
+	return nil
```

Comment:
No need to check the error since we aren't wrapping it after it's been joined.
```suggestion
	return errors.Join(createErrors...)
```

Reply:
fixed